edit/linter-report.js

```diff
@@ -0,0 +1,161 @@
+/* global linter editors clipString createLinterHelpDialog makeSectionVisible */
+'use strict';
+
+var linterReport = (() => { // eslint-disable-line no-var
+  const cms = new Map();
+  const helpDialog = createLinterHelpDialog(getIssues);
+
+  linter.onChange((annotationsNotSorted, annotations, cm) => {
+    let table = cms.get(cm);
+    if (!table) {
+      table = createTable(cm);
+      cms.set(cm, table);
+      const container = $('.lint-report-container');
+      if (typeof editor === 'object') {
+        container.append(table.element);
+      } else {
+        const nextSibling = findNextSibling(cms, cm);
+        container.insertBefore(table.element, nextSibling && cms.get(nextSibling).element);
+      }
+    }
+    table.update();
+    table.updateAnnotations(annotations);
+    updateCount();
+  });
+
+  linter.onUnhook(cm => {
+    const table = cms.get(cm);
+    if (table) {
+      table.element.remove();
+      cms.delete(cm);
+    }
+    updateCount();
+  });
+
+  document.addEventListener('DOMContentLoaded', () => {
+    $('#lint-help').addEventListener('click', helpDialog.show);
+  }, {once: true});
+
+  return {refresh};
+
+  function updateCount() {
+    const issueCount = Array.from(cms.values()).reduce((sum, t) => {
+      sum += t.trs.length;
+      return sum;
+    }, 0);
+    $('#lint').classList.toggle('hidden', issueCount === 0);
+    $('#issue-count').textContent = issueCount;
+  }
+
+  function getIssues() {
+    const issues = new Set();
+    for (const table of cms.values()) {
+      for (const tr of table.trs) {
+        issues.add(tr.getAnnotation());
+      }
+    }
+    return issues;
+  }
+
+  function findNextSibling(cms, cm) {
+    let i = editors.indexOf(cm) + 1;
+    while (i < editors.length) {
+      if (cms.has(editors[i])) {
+        return editors[i];
+      }
+      i++;
+    }
+  }
+
+  function refresh() {
+    for (const table of cms.values()) {
+      table.update();
+    }
+  }
+
+  function createTable(cm) {
```

Comment:
Not sure how to use it. The profiler itself exhausted all memory on my machine (4GB, Windows 7). It shouldn't leak since we make sure the object and the element are removed at the same time:
https://github.com/openstyles/stylus/blob/a7ccda3aadfc7716e59ebb8dc6892645915eb882/edit/linter-report.js#L108